clients/drcachesim/tracer/output.cpp

```diff
@@ -975,7 +975,10 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
         if (op_offline.get_value() && op_split_windows.get_value())
             buf_ptr += instru->append_thread_exit(buf_ptr, dr_get_thread_id(drcontext));
     }
-    if (!skip_size_cap &&
+
+    uintptr_t mode = tracing_mode.load(std::memory_order_acquire);
+
+    if (!skip_size_cap && mode != BBDUP_MODE_L0_FILTER &&
         (is_bytes_written_beyond_trace_max(data) || is_num_refs_beyond_global_max())) {
```

Comment:
Looks like we don't print any message when is_bytes_written_beyond_trace_max() returns true. If it helps in testing, please add a NOTIFY similar to the one below for is_num_refs_beyond_global_max().

Reply:
I agree and I also noticed this, but this seems like a separate issue from the present PR, right?

Reply:
True. I thought of it only when looking at the expected output for one of the added tests. Would be good to have a log that indicates that we indeed hit the max trace bytes limit.

Reply:
I see that the -max_trace_size option is added to tests on Windows. I think we will have to update a number of tests to expect this message on Windows?

```
    macro (torunonly_drcacheoff testname exetgt tracer_ops sim_atops app_args)
    ...
    ...
      if (WIN32 AND NOT ${testname_full}_full_run)
        # Shrink test times on Appveyor.
        set(extra_ops "-max_trace_size 1M")
      endif ()
    ...
```

Reply:
It can be a separate PR if you feel it's better.